server/src/main/java/io/crate/expression/scalar/arithmetic/ExpFunction.java

```diff
@@ -33,19 +33,18 @@ public class ExpFunction {
     public static final String NAME = "exp";
 
     public static void register(Functions.Builder module) {
-        for (var type : DataTypes.NUMERIC_PRIMITIVE_TYPES) {
-            var typeSignature = type.getTypeSignature();
-            module.add(
-                scalar(NAME, typeSignature, typeSignature)
-                    .withFeature(Scalar.Feature.NULLABLE),
-                (declaredSignature, boundSignature) ->
-                    new UnaryScalar<>(
-                        declaredSignature,
-                        boundSignature,
-                        type,
-                        x -> type.sanitizeValue(Math.exp(((Number) x).doubleValue()))
-                    )
-            );
-        }
+        var type = DataTypes.DOUBLE;
```

Comment:
I'm wondering if there is any drawback to register the function with only `DOUBLE` input type, vs registering multiple times with all primitives as input and `DOUBLE` as the return type.
Probably, we add implicit casts for let's say `INTEGER` type to make it match the function signature, but don't know if that's a real issue.

Reply:
FYI, we have a similar function:
https://github.com/crate/crate/blob/ca3b6b9bfc280c6290bbb7f3642ff6514831de5c/server/src/main/java/io/crate/expression/scalar/arithmetic/LogFunction.java#L138-L150

Reply:
:+1: